src/main/java/com/neva/felix/webconsole/plugins/search/core/OsgiExplorer.java

```diff
@@ -154,30 +146,8 @@ public String decompileClass(Long bundleId, String className) {
     }
 
     public String decompileClass(File jar, String className) {
-        String source = StringUtils.EMPTY;
-        String path = StringUtils.replace(className, ".", "/");
-
-        try {
-            final ByteArrayOutputStream stream = new ByteArrayOutputStream();
-            try {
-                try (OutputStreamWriter writer = new OutputStreamWriter(stream)) {
-                    DecompilerSettings settings = DecompilerSettings.javaDefaults();
-                    settings.setTypeLoader(new JarTypeLoader(new JarFile(jar)));
-                    settings.setForceExplicitImports(true);
-                    settings.setForceExplicitTypeArguments(true);
-
-                    Decompiler.decompile(path, new PlainTextOutput(writer), settings);
-                    stream.flush();
-                }
-            } finally {
-                stream.close();
-                source = new String(stream.toByteArray(), Charset.defaultCharset());
-            }
-        } catch (final IOException e) {
-            // handle error
-        }
-
-        return source;
+        LOG.info("jar {} className {}", jar.getAbsolutePath(), className);
```

Comment:
have you compared the output of the same class when using procyon and when using cfr?

Reply:
I don't have any local setup to use a previous version of java and have procyon working again. I'll see what I can try.